backend/account/views.py

```diff
@@ -22,14 +23,46 @@ def get_or_create_account(request: HttpRequest):
     Creates a new account, given google id token.
     Will use the google id token to get the user's information.
     """
+    logger.debug("Here!")
+    if not request.COOKIES.get("csrftoken"):
+        return HttpResponse("No CSRF Token in Cookie", status=401)
+    elif not request.headers.get("X-Csrftoken"):
+        return HttpResponse("No CSRF Token in Header", status=401)
+    elif request.COOKIES.get("csrftoken") != request.headers.get("X-Csrftoken"):
+        return HttpResponse("CSRF Validation Failed", status=401)
 
     body = read_request(request)
-    credential = body['credential']
+    client_id = body["client_id"]
+    credential = body["credential"]
     logger.debug(f"get_or_create_account: {credential}")
+    logger.debug(f"body: {body}")
+
+    # Verify Credentials via Google
+    try:
+        # Specify the CLIENT_ID of the app that accesses the backend:
+        idinfo = id_token.verify_oauth2_token(
+            credential, requests.Request(), client_id, clock_skew_in_seconds=5
+        )
+
+        # ID token is valid. Get the user's Google Account ID from the decoded token.
+        if idinfo["sub"]:
+            logger.debug(f"Credential Validated for User.google_id: { idinfo['sub'] }")
 
-    user = business.get_or_create_user({'google_id': credential})
+            user = business.get_or_create_user(idinfo)
+            logger.debug(f"Returned user:\n{user.to_dict()}")
+            # Need some way to differenciate first-time logins
+            # check user.last_login in frontend
+            # if None -> first login, redirect to account setup
+            # regardless, post to login_user endpoint to set value?
+            return JsonResponse({"data": user.to_dict()})
+        else:  # probably removable
```

Comment:
You can probably remove this (and the if statement on line 48) - or you could change line 48 to `if idinfo.get("sub")` and remove the try-except

Reply:
removed the else. Keeping the try-except due to verify_oauth2_token functionality